api.planx.uk/helpers.ts

```diff
@@ -66,55 +66,96 @@ interface CreateFlowResponse {
   };
 }
 
+const createFlowMutation = gql`
+  mutation InsertFlow(
+    $team_id: Int!
+    $slug: String!
+    $name: String!
+    $data: jsonb = {}
+    $copied_from: uuid
+    $templated_from: uuid
+  ) {
+    flow: insert_flows_one(
+      object: {
+        team_id: $team_id
+        slug: $slug
+        name: $name
+        data: $data
+        version: 1
+        copied_from: $copied_from
+        templated_from: $templated_from
+      }
+    ) {
+      id
+    }
+  }
+`;
+
+const createSourceTemplateMutation = gql`
+  mutation InsertSourceTemplateFlow(
+    $team_id: Int!
+    $slug: String!
+    $name: String!
+    $data: jsonb = {}
+    $copied_from: uuid
+    $templated_from: uuid
+    $is_template: Boolean
+  ) {
+    flow: insert_flows_one(
+      object: {
+        team_id: $team_id
+        slug: $slug
+        name: $name
+        data: $data
+        version: 1
+        copied_from: $copied_from
+        templated_from: $templated_from
+        is_template: $is_template
```

Comment:
If this is the `createSourceTemplateMutation`, does `is_template` actually have to be passed in as an argument or can we always default to `true`? 

Would this help address the pattern below at all? I'm thinking something like:
- `createFlow` still accepts an `isTemplate` argument
- But we keep a single mutation and simply set the correct `isTemplate` graphql param ourselves? (defaulting to false?)

Does that still work with Hasura permissions to restrict creation of source templates to platform admins?

Reply:
> But we keep a single mutation and simply set the correct isTemplate graphql param ourselves? (defaulting to false?)

This was what I wanted to go for and tried first of all - non `platformAdmins` can't even write `false` to this column however, it needs to be a column they can't interact with. I tried a few GraphQL approaches with `@include` for this variable and didn't manage to get a working version without two different mutations.

The other approach here could be either - 
 - we allow anybody to write to this column (not ideal, but is it actually that risky?)
 - we allow anybody to write to the column, but then validate their permission after with a post-update check (would work, but a fair bit more complex)

Reply:
Thanks for explanation ! I'd say either accept the risk of any role writing to column (we know that non-OSL authored templates are probably a near future feature anyways - eg RAB), or keep this approach for now (seems better than more complex post-update check?)

Reply:
Yep good shout on non-OSL templates, updated permissions and simplified mutation - much better!